src/ui/simulator/toolbox/components/htmllistbox/item/cluster-item.cpp

```diff
@@ -60,24 +60,30 @@ void ClusterItem::preloadImages()
     {
         pIconFileEnabled = getIconFilePath("images/16x16/light_green.png");
         pIconFileDisabled = getIconFilePath("images/16x16/light_orange.png");
-        pClusterIconFilePath = getClusterIconFilePath();
     }
 }
 
+void ClusterItem::addAdditionalIcons(wxString& out) const
```

Comment:
An empty method is a code smell. **addAdditionalIcons(...)** should be a private function of class **ThermalClusterItem**.
To do this, we could : 
- split **ClusterItem::HtmlContent(...)** into **ClusterItem**'s protected functions with appropriate names,
- make **ClusterItem::HtmlContent(...)** abstract
- define a **HtmlContent** method for each **ClusterItem**'s child (thermal or renewable), and call the **ClusterItem**'s protected functions depending on the child. **ThermalClusterItem** could call its own **addAdditionalIcons**.

Reply:
It's a code smell if you leave the body of the function empty, which isn't the case here.

I agree that we could split functions, but we would lose a lot of  expressiveness. How is the child supposed to implement `HtmlContent_part1`, `HtmlContent_part2` and so on ?

We don't want to stick too much on the implementation of the base class here.